source/Applications/Advanced/PoseConversions/PoseConversions.cs

```diff
@@ -0,0 +1,325 @@
+﻿/*
+This example shows how to perform Hand-Eye calibration.
+*/
+
+using System;
+using System.IO;
+using YamlDotNet.RepresentationModel;
+
+
+using MathNet.Numerics.LinearAlgebra;
+using MathNet.Numerics.LinearAlgebra.Double;
+
+class Program
+{
+    static void Main()
+    {
+        try
+        {
+            var zivid = new Zivid.NET.Application();
+
+            printHeader("This example shows conversions to/from Transformation Matrix");
+
+            // Read a transformation matrix
+            var transformationMatrixFile = Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData)
+                               + "/Zivid/RobotTransform.yaml";
+            Console.WriteLine("Getting PoseState:");
+            var transformationMatrix = readTransform(transformationMatrixFile);
+            var mathMatrix = zividToMathDotNet(transformationMatrix);
+            Console.WriteLine(matrixToString(mathMatrix));
+
+            // Extract rotation matrix and translation vector
+            var rotationMatrix = mathMatrix.SubMatrix(0, 3, 0, 3);
+            var translationVector = mathMatrix.SubMatrix(0, 3, 3, 1);
+
+            Console.WriteLine("RotationMatrix:\n" + matrixToString(rotationMatrix));
```

Comment:
I would remove the empty line here to keep extracting and printing these two together.

Reply:
In the C++ sample, we don't have variables for rotationMatrix and translationVector. For consistency, you would want to put mathMatrix.SubMatrix(0, 3, 0, 3) in lines 42, 46, 48, etc. But perhaps extracting these two, makes it easier to read and follow the code. Is this why you decided to do it? If so, we should consider improving the C++ variant.

Reply:
I think I read a bit fast over the C++ code then. In the comment in C++ we write: "Extract the rotation matrix and translation vector...", so I assumed we did. Maybe change the comment to "print out rotation and translation..." instead of "extract rotation and translation..." since we are not extracting them?